### PR TITLE
Kogito 3568 - Remove spec.httpPort attribute from Operator and HTTP_PORT env var from Kogito images

### PR DESCRIPTION
**Pull request by Kaitou786**
See: https://issues.redhat.com/browse/KOGITO-3568
Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: modules/kogito-jobs-service/added/kogito-app-launch.sh [hunk 1]
```diff
@@ -20,4 +20,5 @@ source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
 exec java ${SHOW_JVM_SETTINGS} ${JAVA_OPTIONS} ${KOGITO_JOBS_PROPS} \
```
Comment (spolti) [inline modules/kogito-jobs-service/added/kogito-app-launch.sh:22]: can you please add -Dquarkus.http.host=0.0.0.0  here?

File: modules/kogito-quarkus-s2i/added/kogito-app-launch.sh [hunk 2]
```diff
@@ -14,15 +14,16 @@ fi
...
 
 if [ "${NATIVE^^}" == "TRUE" ]; then
-    exec $KOGITO_HOME/bin/*-runner ${JAVA_OPTIONS} ${KOGITO_QUARKUS_S2I_PROPS} \
+    exec $KOGITO_HOME/bin/*-runner ${JAVA_OPTIONS} \
          -Dquarkus.http.host=0.0.0.0 -Djava.library.path=$KOGITO_HOME/ssl-libs \
-        -Djavax.net.ssl.trustStore=$KOGITO_HOME/cacerts
+         -Dquarkus.http.port=8080 \
+         -Djavax.net.ssl.trustStore=$KOGITO_HOME/cacerts
 else
     exec java ${JAVA_OPTIONS} ${KOGITO_QUARKUS_S2I_PROPS} -Dquarkus.http.host=0.0.0.0 -jar $KOGITO_HOME/bin/*runner.jar
```
Comment (spolti) [inline modules/kogito-quarkus-s2i/added/kogito-app-launch.sh]: please add the port on the else clause as well.

File: modules/kogito-springboot-s2i/added/kogito-app-launch.sh [hunk 3]
```diff
@@ -15,10 +15,10 @@ fi
...
 # Any configuration script that needs to run on image startup must be added here.
 CONFIGURE_SCRIPTS=(
-  ${KOGITO_HOME}/launch/kogito-springboot-s2i.sh
+
 )
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
-exec java ${JAVA_OPTIONS} ${KOGITO_SPRINGBOOT_S2I_PROPS} -jar $KOGITO_HOME/bin/*.jar
+exec java ${JAVA_OPTIONS} -Dserver.port=8080 -jar $KOGITO_HOME/bin/*.jar
```
Comment (spolti) [inline modules/kogito-springboot-s2i/added/kogito-app-launch.sh]: please add the bind address here

File: modules/kogito-springboot/added/kogito-app-launch.sh [hunk 4]
```diff
@@ -15,9 +15,9 @@ fi
...
 # Any configuration script that needs to run on image startup must be added here.
 CONFIGURE_SCRIPTS=(
-  ${KOGITO_HOME}/launch/kogito-springboot.sh
+
 )
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
-exec java ${JAVA_OPTIONS} ${KOGITO_SPRINGBOOT_PROPS} -jar $KOGITO_HOME/bin/*.jar
+exec java ${JAVA_OPTIONS} -Dserver.port=8080 -jar $KOGITO_HOME/bin/*.jar
```
Comment (spolti) [inline modules/kogito-springboot/added/kogito-app-launch.sh]: please add the bind address

File: tests/features/kogito-data-index.feature [hunk 5]
```diff
@@ -0,0 +1,54 @@
+@quay.io/kiegroup/kogito-data-index
```
Comment (spolti) [inline tests/features/kogito-data-index.feature]: any reason to re-add it?

**Review by spolti — CHANGES_REQUESTED** [review]
well done!, just a few things.

File: tests/features/kogito-data-index.feature (hunk 5 above)
Comment (Kaitou786) [reply tests/features/kogito-data-index.feature]: Ah, no I think I messed up while rebasing with the latest changes :(

**Review by vaibhavjainwiz — APPROVED** [review]
LGTM

**Review by sutaakar — APPROVED** [review]

File: modules/kogito-quarkus/added/kogito-app-launch.sh [hunk 6]
```diff
@@ -15,11 +15,12 @@ fi
 # Configuration scripts
 # Any configuration script that needs to run on image startup must be added here.
 CONFIGURE_SCRIPTS=(
-  ${KOGITO_HOME}/launch/kogito-quarkus-native.sh
+
 )
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
-exec $KOGITO_HOME/bin/*-runner ${JAVA_OPTIONS} ${KOGITO_QUARKUS_NATIVE_PROPS} \
```
Comment (radtriste) [inline modules/kogito-quarkus/added/kogito-app-launch.sh:23]: why removing `KOGITO_QUARKUS_NATIVE_PROPS` here ?

File: modules/kogito-quarkus-jvm/added/kogito-app-launch.sh [hunk 7]
```diff
@@ -15,9 +15,12 @@ fi
 # Configuration scripts
 # Any configuration script that needs to run on image startup must be added here.
 CONFIGURE_SCRIPTS=(
-  ${KOGITO_HOME}/launch/kogito-quarkus-jvm.sh
+
 )
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
-exec java ${JAVA_OPTIONS} ${KOGITO_QUARKUS_JVM_PROPS} -Dquarkus.http.host=0.0.0.0 -jar $KOGITO_HOME/bin/*runner.jar
```
Comment (radtriste) [inline modules/kogito-quarkus-jvm/added/kogito-app-launch.sh:23]: why removing `KOGITO_QUARKUS_JVM_PROPS` ?

File: modules/kogito-quarkus-s2i/added/kogito-app-launch.sh [hunk 8]
```diff
@@ -14,15 +14,16 @@ fi
...
 # Any configuration script that needs to run on image startup must be added here.
 CONFIGURE_SCRIPTS=(
-  ${KOGITO_HOME}/launch/kogito-quarkus-s2i.sh
+
 )
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
 if [ "${NATIVE^^}" == "TRUE" ]; then
-    exec $KOGITO_HOME/bin/*-runner ${JAVA_OPTIONS} ${KOGITO_QUARKUS_S2I_PROPS} \
```
Comment (radtriste) [inline modules/kogito-quarkus-s2i/added/kogito-app-launch.sh:23]: why removing `KOGITO_QUARKUS_S2I_PROPS` ?

File: modules/kogito-springboot-s2i/added/kogito-app-launch.sh [hunk 9]
```diff
@@ -15,10 +15,10 @@ fi
 # Configuration scripts
 # Any configuration script that needs to run on image startup must be added here.
 CONFIGURE_SCRIPTS=(
-  ${KOGITO_HOME}/launch/kogito-springboot-s2i.sh
+
 )
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
-exec java ${JAVA_OPTIONS} ${KOGITO_SPRINGBOOT_S2I_PROPS} -jar $KOGITO_HOME/bin/*.jar
```
Comment (radtriste) [inline modules/kogito-springboot-s2i/added/kogito-app-launch.sh:23]: why removing `KOGITO_SPRINGBOOT_S2I_PROPS` ?

File: modules/kogito-springboot/added/kogito-app-launch.sh [hunk 10]
```diff
@@ -15,9 +15,9 @@ fi
 # Configuration scripts
 # Any configuration script that needs to run on image startup must be added here.
 CONFIGURE_SCRIPTS=(
-  ${KOGITO_HOME}/launch/kogito-springboot.sh
+
 )
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
-exec java ${JAVA_OPTIONS} ${KOGITO_SPRINGBOOT_PROPS} -jar $KOGITO_HOME/bin/*.jar
```
Comment (radtriste) [inline modules/kogito-springboot/added/kogito-app-launch.sh:23]: why removing `KOGITO_SPRINGBOOT_PROPS` ?

**Review by radtriste — COMMENTED** [review]
comments about env vars no more used for runtime/s2i builds ?
(btw those are left for services)

File: modules/kogito-quarkus/added/kogito-app-launch.sh (hunk 6 above)
Comment (Kaitou786) [reply modules/kogito-quarkus/added/kogito-app-launch.sh:23]: They are not used anywhere in the module now. I left it for the services as user can define custom flags or props for services using that env, not sure if user wants to define any custom flags for runtime/builder images.

Comment (radtriste) [reply modules/kogito-quarkus/added/kogito-app-launch.sh:23]: I think we should leave that possibility @spolti @sutaakar  ?

**Review by ricardozanini — APPROVED** [review]
LGTM once @radtriste comments are handled.

File: modules/kogito-quarkus/added/kogito-app-launch.sh (hunk 6 above)
Comment (sutaakar) [reply modules/kogito-quarkus/added/kogito-app-launch.sh:23]: I think we can keep it.

Comment (Kaitou786) [reply modules/kogito-quarkus/added/kogito-app-launch.sh:23]: Added the props back :)

**Review by radtriste — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Pull request merged**